### ISSUE TITLE
Get and parse to a generic T inferred from return T

### ISSUE THREAD

**Issue by lweberk** [sagiegurari/envmnt#26]
### Feature Description
A function to access and parse into any type T or error out.

### Describe The Solution You'd Like
A function `envmnt::get<T>() -> Result<T, ?> where T: FromStr<T>`. This way a environment variable can be gotten and parsed by its return type, or gracefully error out. 

### Code Sample
```rust
pub fn get<T, E>(key: &'static str) -> EnvmtResult<T>
    where T: std::str::FromStr + std::str::FromStr<Err=E>,
          E: std::fmt::Display
{
    let var = std::env::var(key)
        .map_err(|e|
            EnvmtError::Missing(format!("Failed to load environment variable: '{}' ({})", name, e))
        )?;

    let val = var.parse::<T>()
        .map_err(|e|
            EnvmtError::InvalidType(
                format!("Failed to parse environment variable: '{}' to type '{}' ({})", name, stringify!(T), e)
            )
        )?;

    Ok(val)
}
```

### Collateral Effects

For instance numeric.rs can be simplified by this, since this generic specializes amongst other things into all its variants.

### Motivation

I'm currently carrying this boiler plate around project after project, so I figured it might be useful to upstream it into an existing env utilities crate like yours. Would you accept a pull request for something like this?

**Comment by sagiegurari** [comment]
definitely happy to get a PR.
as for numeric.rs its all macros so it feels simple already, but lets see

**Comment by sagiegurari** [comment]
This is now released in 0.9 version
